src/ShellCheck/Checks/Commands.hs

```diff
@@ -620,6 +622,9 @@ checkInteractiveSu = CommandCheck (Basename "su") f
     undirected (T_Redirecting _ (_:_) _) = False
     undirected _ = True
 
+    noCommandArgPresent = not . any isCommandArg . map (getLiteralStringDef "_")
+    isCommandArg arg = arg == "-c" && "--command=" `isPrefixOf` arg
```

Comment:
This is always false because of the `&&`, and would fail to recognize `--command foo` (without `=`) since it would now apply to multiple arguments as well, and something like `su -Pc cmd` is valid if rare.